src/io/browser/browser-io.ts

```diff
@@ -31,6 +35,32 @@ export function makeBrowserIo(): EdgeIo {
     // Networking:
     fetch(uri: string, opts?: EdgeFetchOptions): Promise<EdgeFetchResponse> {
       return window.fetch(uri, opts)
+    },
+    async fetchCors(
+      uri: string,
+      opts?: EdgeFetchOptions
+    ): Promise<EdgeFetchResponse> {
+      const { hostname } = new URL(uri)
+
+      // Proactively use fetchCorsProxy for any hostnames added to whitelist:
+      if (hostnameProxyWhitelist.has(hostname)) {
+        return await fetchCorsProxy(uri, opts)
+      }
+
+      try {
+        // Attempt regular fetch:
+        return await window.fetch(uri, opts)
+      } catch (error) {
+        // Fallback to edge-core-proxy server if this is a CORS error:
+        if (String(error) === 'TypeError: Load failed') {
```

Comment:
Not all errors messages are the same across all platforms, so we should assume all errors are potentially CORS related and try the cors proxy fetch.

Reply:
Also consider a Map of the failure counts to assume errors aren't to do with CORS